src/Ship.cpp

```diff
@@ -1453,15 +1460,48 @@ void Ship::Render(Graphics::Renderer *renderer, const Camera *camera, const vect
 	}
 }
 
-bool Ship::SpawnCargo(CargoBody *c_body) const
+bool Ship::SpawnCargo(CargoBody *c_body)
 {
-	if (m_flightState != FLYING) return false;
-	vector3d pos = GetOrient() * vector3d(0, GetAabb().min.y - 5, 0);
+	if (m_flightState != FLYING) {
+		// cleanup the allocated CargoBody
+		delete c_body;
+		return false;
+	}
+
+	// don't actually spawn the cargo just yet, add it to a queue for spawning later
+	m_cargoSpawnQueue.push_back(c_body);
+	return true;
+}
+
+void Ship::ProcessSpawnQueue()
+{
+	// check we're due to spawn something
+	const double latestTime = Pi::game->GetTime();
+	if (m_latestSpawnTime > latestTime)
+		m_latestSpawnTime = 0.0; // don't allow time to be in the future, reset it
+
+	if (m_latestSpawnTime + 0.2 <= latestTime) {
```

Comment:
Hardcoded spawn interval - this entire method should ideally be moved to a Lua timer IMO with a single API `bool Ship::SpawnCargo(...)` which returns false if the area to stick the cargo in isn't clear.

Reply:
I could not figure out what to do with the Lua, and it kinda requires the whole interface be different. You'll never be able to spawn them, there'd have be collision checking etc. Just way beyond anything I have time to do or the Lua ability right now

Reply:
That's fine, I was mentioning it as the "ideal scenario" when all of the pieces of the puzzle are in place. This is a good enough compromise for now I think.